addons/todo/static/src/views/todo_form/todo_form_controller.js

```diff
@@ -1,14 +1,55 @@
 /** @odoo-module **/
 
+import { _lt } from "@web/core/l10n/translation";
 import { FormController } from "@web/views/form/form_controller";
 import { TodoEditableBreadcrumbName } from "@todo/components/todo_editable_breadcrumb_name/todo_editable_breadcrumb_name";
+const { onWillStart } = owl;
 
 /**
  *  The FormController is overridden to be able to manage the edition of the name of a to-do directly
  *  in the breadcrumb.
  */
 
-export class TodoFormController extends FormController {}
+export class TodoFormController extends FormController {
+    setup() {
+        super.setup();
+        onWillStart(async () => {
+            this.conversionViewId = await this.model.orm.call(
+                'project.task',
+                'get_conversion_view_id',
+                [],
+            );
+        });
+    }
+
+    get actionMenuItems() {
+        this.archiveEnabled = true;
+        const actionToKeep = ["archive", "unarchive", "duplicate", "delete"];
+        const printToKeep = [];
+        const menuItems = super.actionMenuItems;
+        const filteredActions = menuItems.action?.filter(action => actionToKeep.includes(action['key'])) || [];
+        const filteredPrints = menuItems.print?.filter(print => printToKeep.includes(print['key'])) || [];;
+
+        if (this.conversionViewId) {
+            filteredActions.push({
+                description: _lt("Convert to Task"),
+                callback: () => {
+                    this.model.actionService.doAction({
+                        type: 'ir.actions.act_window',
+                        name: _lt('Convert to Task'),
+                        target: 'new',
+                        res_model: 'project.task',
+                        res_id: this.model.root.data.id,
+                        views: [[this.conversionViewId, "form"]],
+                    });
```

Comment:
Another solution could be to call the action defined in python and that action will return a dict containing all the information you passed here in the JS object. 🤔

Reply:
I tried this solution but it doesn't seem doable, or at least would not result in a better implementation than the actual. I need to create my server action in xml, but then if I want to call it from the callback I still need to do an rpc call to get its id.